beacon-chain/cache/depositsnapshot/zero_node.go

```diff
@@ -0,0 +1,39 @@
+package depositsnapshot
+
+import (
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v3/crypto/hash"
+)
+
+// Zero represents an empty node without a deposit and satisfies the MerkleTreeNode interface.
+type Zero struct {
+	depth uint64
+}
+
+// GetRoot returns the root of the Merkle tree.
+func (z *Zero) GetRoot() [32]byte {
+	if z.depth == DepositContractDepth {
+		return hash.Hash(append(zeroHash[:], zeroHash[:]...))
+	}
+	return zeroHash
+}
+
+// IsFull returns whether there is space left for deposits.
+func (z *Zero) IsFull() bool {
+	return false
+}
```

Comment:
Same feedback as leaf node. Add commentary why this always returns false.

Reply:
Same as above, additional commentary added, if you could please check that it is clear.